src/silx/app/compare/CompareImagesWindow.py

```diff
@@ -27,21 +27,37 @@
 
 import logging
 import numpy
+import typing
+import os.path
 
 import silx.io
 from silx.gui import icons
 from silx.gui import qt
-from silx.io.url import DataUrl
 from silx.gui.plot.CompareImages import CompareImages
 from silx.gui.widgets.UrlSelectionTable import UrlSelectionTable
 from ..utils import parseutils
 from silx.gui.plot.tools.profile.manager import ProfileManager
 from silx.gui.plot.tools.compare.profile import ProfileImageDirectedLineROI
 
+try:
+    import PIL
+except ImportError:
+    PIL = None
```

Comment:
Since `fabio` already has a dependency on `pillow`, adding `pillow` as a dependency of `silx` won't change its effective dependencies and would make it simpler here.